dc_logging_aws/named_queries/commands/create_election_query_files.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import sys
+from datetime import datetime, time
+from pathlib import Path
+
+# Add the parent directory to the Python path
+script_dir = Path(__file__).resolve().parent
+project_root = script_dir.parent.parent.parent
+sys.path.insert(0, str(project_root))
+
+from dc_logging_aws.named_queries.query_template import (  # noqa: E402
+    QueryTemplate,
+)
+
+
+class QueryFileCreator:
+    def __init__(self, polling_day, start_of_election_period, overwrite=False):
+        self.polling_day = polling_day
+        self.start_of_election_period = start_of_election_period
+        self.overwrite = overwrite
+        self.date_str = polling_day.strftime("%Y-%m-%d")
+        self.query_template = QueryTemplate(
+            polling_day, start_of_election_period
+        )
+        self.script_dir = Path(__file__).resolve().parent
+
+    @staticmethod
+    def valid_date(s):
+        try:
+            return datetime.strptime(s, "%Y-%m-%d").date()
+        except ValueError:
+            msg = f"Not a valid date: '{s}'. Please use YYYY-MM-DD format."
+            raise argparse.ArgumentTypeError(msg)
+
+    def create_query_directory(self):
+        queries_dir = self.script_dir.parent / "queries"
```

Comment:
The outstanding question I have is whether we want the `queries` directory in the gh repo or not. I think I err to 'yes', because it's nice to keep a record of these things, and it's easy to delete things in Athena. But maybe it should be 'no' until we're actually using CI (or some other automation to) to run the queries. This is because we'll have 2 sources of truth (athena and gh) until that's the case...